lib/src/main/java/com/diffplug/spotless/java/PalantirJavaFormatStep.java

```diff
@@ -0,0 +1,152 @@
+ * <a href="https://github.com/google/google-java-format">google-java-format</a> as a FormatterStep. */
+public class PalantirJavaFormatStep {
+	// prevent direct instantiation
+	private PalantirJavaFormatStep() {}
+
+	private static final String DEFAULT_STYLE = "PALANTIR";
+	static final String NAME = "palantir-java-format";
+	static final String MAVEN_COORDINATE = "com.palantir.javaformat:palantir-java-format";
+	static final String FORMATTER_CLASS = "com.palantir.javaformat.java.Formatter";
+	static final String FORMATTER_METHOD = "formatSource";
+
+	private static final String OPTIONS_CLASS = "com.palantir.javaformat.java.JavaFormatterOptions";
+	private static final String OPTIONS_BUILDER_METHOD = "builder";
+	private static final String OPTIONS_BUILDER_CLASS = "com.palantir.javaformat.java.JavaFormatterOptions$Builder";
+	private static final String OPTIONS_BUILDER_STYLE_METHOD = "style";
+	private static final String OPTIONS_BUILDER_BUILD_METHOD = "build";
+	private static final String OPTIONS_STYLE = "com.palantir.javaformat.java.JavaFormatterOptions$Style";
+
+	private static final String REMOVE_UNUSED_CLASS = "com.palantir.javaformat.java.RemoveUnusedImports";
+	private static final String REMOVE_UNUSED_METHOD = "removeUnusedImports";
+
+	private static final String IMPORT_ORDERER_CLASS = "com.palantir.javaformat.java.ImportOrderer";
+	private static final String IMPORT_ORDERER_METHOD = "reorderImports";
+
+	/** Creates a step which formats everything - code, import order, and unused imports. */
+	public static FormatterStep create(Provisioner provisioner) {
+		return create(defaultVersion(), provisioner);
+	}
+
+	/** Creates a step which formats everything - code, import order, and unused imports. */
+	public static FormatterStep create(String version, Provisioner provisioner) {
+		return create(version, DEFAULT_STYLE, provisioner);
+	}
+
+	/** Creates a step which formats everything - code, import order, and unused imports. */
+	public static FormatterStep create(String version, String style, Provisioner provisioner) {
+		return create(MAVEN_COORDINATE, version, style, provisioner);
+	}
+
+	/** Creates a step which formats everything - groupArtifact, code, import order, and unused imports. */
+	public static FormatterStep create(String groupArtifact, String version, String style, Provisioner provisioner) {
+		Objects.requireNonNull(groupArtifact, "groupArtifact");
+		if (groupArtifact.chars().filter(ch -> ch == ':').count() != 1) {
+			throw new IllegalArgumentException("groupArtifact must be in the form 'groupId:artifactId'");
+		}
+		Objects.requireNonNull(version, "version");
+		Objects.requireNonNull(style, "style");
+		Objects.requireNonNull(provisioner, "provisioner");
+		return FormatterStep.createLazy(NAME,
+				() -> new State(NAME, groupArtifact, version, style, provisioner),
+				State::createFormat);
+	}
+
+	static final Jvm.Support<String> JVM_SUPPORT = Jvm.<String> support(NAME).add(8, "1.1.0").add(11, "2.10.0");
```

Comment:
If you killed `PalantirJavaFormatStep`, this could instead be a field `PALANTIR_JVM_SUPPORT` inside `GoogleJavaFormatStep`.

Reply:
My preference is to keep the two implementations separate, coupling is more likely to cause friction than to be helpful if things diverge. Supporting large matrices across version ranges and separate artifacts will make both harder to support.

Reply:
Sounds good to me :)